a3p-integration/proposals/z:acceptance/test-lib/psm-lib.js

```diff
@@ -0,0 +1,492 @@
+ *   newParams: Params
+ *   deadline: number
+ *   offerId?: string
+ * }} question
+ *
+ * @param {{
+ *   committeeAddrs: Array<string>
+ *   position: number
+ * }} voting
+ */
+export const implementPsmGovParamChange = async (question, voting) => {
+  await buildProposePSMParamChangeOffer(question);
+  await voteForNewParams(voting);
+  console.log('ACTIONS wait for the vote deadline to pass');
+  await new Promise(r => setTimeout(r, VOTING_WAIT_MS));
+};
+
+export const checkGovParams = async (t, expected, psmName) => {
+  const current = await getPsmGovernance(psmName);
+
+  t.log({
+    give: current.WantMintedFee.value,
+    want: current.GiveMintedFee.value,
+    mintLimit: current.MintLimit,
+  });
+
+  t.like(current, expected);
+};
+
+/**
+ *
+ * @param {string} name
+ * @param {{
+ *   denom: string,
+ *   value: string
+ * }} fund
+ */
+export const initializeNewUser = async (name, fund) => {
+  const psmTrader = await addUser(name);
+  await Promise.all([
+    bankSend(psmTrader, `20000000ubld,${fund.value}${fund.denom}`),
+    bankSend(psmTrader, `1000000uist`, GOV1ADDR),
+  ]);
+};
+
+/**
+ *
+ * @param {any} t
+ * @param {string} userName
+ * @param {{
+ *   denom: string,
+ *   value: string
+ * }} expectedAnchorFunds
+ */
+export const checkUserInitializedSuccessfully = async (
+  t,
+  userName,
+  expectedAnchorFunds,
+) => {
+  let userAddress;
+  await t.notThrowsAsync(async () => {
+    userAddress = await getUser(userName);
+  });
+
+  // @ts-expect-error initialization
+  const balance = await getBalances([userAddress], expectedAnchorFunds.denom);
+  t.is(balance >= BigInt(expectedAnchorFunds.value), true);
+};
+
+/**
+ * Similar to https://github.com/Agoric/agoric-3-proposals/blob/422b163fecfcf025d53431caebf6d476778b5db3/packages/synthetic-chain/src/lib/commonUpgradeHelpers.ts#L123-L139
+ * However, in the case where "address" is not provisioned "agoric wallet send" is needed because
+ * "agops perf satisfaction" tries to follow ":published.wallet.${address}" which blocks the execution because no such path exists in
+ * vstorage. In situations like this "agoric wallet send" seems a better choice as it doesn't depend on following user's vstorage wallet path
+ *
+ * @param {string} address
+ * @param {Promise<string>} offerPromise
+ */
+export const sendOfferAgoric = async (address, offerPromise) => {
+  const offerPath = await mkTemp('agops.XXX');
+  const offer = await offerPromise;
+  await fsp.writeFile(offerPath, offer);
+
+  await agoric.wallet(
+    '--keyring-backend=test',
+    'send',
+    '--offer',
+    offerPath,
+    '--from',
+    address,
+  );
+};
+
+/**
+ * @param {string} address
+ * @param {Array<any>} params
+ */
+export const agopsPsm = (address, params) => {
+  const newParams = ['psm', ...params];
+  const offerPromise = executeCommand(agopsLocation, newParams);
+  return sendOfferAgoric(address, offerPromise);
+};
+
+const giveAnchor = (base, fee) => {
+  return Math.ceil(base / (1 - fee));
+};
```

Comment:
consider more concise for these math expressions:
```suggestion
const giveAnchor = (base, fee) => Math.ceil(base / (1 - fee));
```

Reply:
resolved in https://github.com/Agoric/agoric-sdk/commit/712f97060737ce2a54d4e27ba2047142da6585ab